tools/checklocks/state.go

```diff
@@ -184,7 +184,46 @@ func (l *lockState) count() int {
 
 // isCompatible returns true if the states are compatible.
```

Comment:
What does compatible mean?

Reply:
Why is nil compatible with anything else?

Make sure documentation answers key questions like this.